tools/protodoc/protodoc.py

```diff
@@ -160,6 +160,12 @@ def format_comment_with_annotations(comment, type_name=''):
     Returns:
         A string with additional RST from annotations.
     """
+    alpha_warning = ''
+    if annotations.ALPHA_ANNOTATION in comment.annotations:
+        experimental_warning = (
+            '.. warning::\n   This API is alpha and is not covered by the :ref:`threat model <arch_overview_threat_model>`.\n\n'
```

Comment:
FWIW, we have a broader issue which we can resolve after we land this PR. We have both implementation and API alpha/experimental/unknown robustness status. An API is not really insecure in and of itself, this is an implementation attribute.

We really need a way to mark in the protos client specific implementation status details, since Envoy vs. gRPC support for a feature might have different implementation status.

I started to explore this in https://github.com/envoyproxy/envoy/issues/11085, but it's not really something I have b/w for. Maybe @phlax this can be part of the docs/tooling roadmap?

CC  @markdroth

Reply:
I agree, it would be nice to have some generic solution for this kind of thing.  Matt has been saying for a while that we need a way to have client-specific docs in the proto files, not just for API status but for any type of documentation.  Although as the number of xDS clients grows, I'm not sure how feasible it will be to document all of the in the same proto files.  But this is definitely something we need to come up with a good strategy for.